src/commands/aksDeployManifest/aksDeployManifest.ts

```diff
@@ -0,0 +1,145 @@
+
+const GITHUBCOPILOT_FOR_AZURE_VSCODE_ID = "ms-azuretools.vscode-azure-github-copilot";
+
+export async function aksDeployManifest() {
+    // Check if GitHub Copilot for Azure extension is installed
+    const checkGitHubCopilotExtension = checkExtension(GITHUBCOPILOT_FOR_AZURE_VSCODE_ID);
+
+    if(!checkGitHubCopilotExtension) {
+        handleExtensionDoesNotExist(GITHUBCOPILOT_FOR_AZURE_VSCODE_ID);
+        return;
+    }
+
+    // Select manifest
+    const manifest = await getManifestFile();
+
+    if(manifest === undefined) { 
+        return;
+    }
+
+    // Select cluster
+    const sessionProvider = await getReadySessionProvider();
+
+    if(failed(sessionProvider)) {
+        vscode.window.showErrorMessage(sessionProvider.error);
+        return;
+    }
+
+    const cluster = await selectClusterOptions(sessionProvider.result);
+
+    if (failed(cluster)) {
+        vscode.window.showErrorMessage(cluster.error);
+        return;
+    }
+
+    if (cluster.result === false) {
+        vscode.window.showWarningMessage("No cluster selected.");
+        return;
+    }
+
+    const clusterPreference = cluster.result as ClusterPreference;
+
+    // Confirm deployment
+    const confirmed = await confirmDeployment(clusterPreference.clusterName);
+
+    if(!confirmed) {
+        vscode.window.showWarningMessage("Deployment cancelled.");
+        return;
+    }
+
+    const deploymentResult = await deployApplicationToCluster({
+        cluster: clusterPreference,
+        manifestPath: manifest
+    });
+
+    if(failed(deploymentResult)) {
+        vscode.window.showErrorMessage(deploymentResult.error);
+        return;
+    }
+
+    vscode.window.showInformationMessage(`Your applicatoin has been successfully deployed to the ${clusterPreference.clusterName} AKS cluster.`, "View resource in the Azure portal").then((res) => {
+        if(res) {
+            vscode.env.openExternal(vscode.Uri.parse(deploymentResult.result.url));
+        }
+    });
+}
+
+type ManifestQuickPickItem = vscode.QuickPickItem & {filePath: string};
+
+async function getManifestFile(): Promise<string | undefined> {
+    const items: ManifestQuickPickItem[] = [];
+    await vscode.workspace.findFiles(`**/**.yaml`, "**/node_modules/**").then((result) => {
+        result.forEach((fileUri) => {
+            const fileName = path.basename(fileUri.fsPath);
+            items.push({ label: fileName, description: fileUri.fsPath, filePath: fileUri.fsPath });
+        });
+    });
+
+    if(items.length === 0) {
+        vscode.window.showWarningMessage("No manifest files found in the workspace.");
+        return;
+    }
+
+    const fileSelected = await vscode.window.showQuickPick(items.sort(), {
+        title: "Select YAML",
+        placeHolder: "Select manifest to deploy",
+    });
+
+    if (!fileSelected) {
+        vscode.window.showWarningMessage("Manifest file not selected");
+        return;
+    }
+
+    return fileSelected.filePath;
+}
```

Comment:
💡 Suggestion/observation: for little rewrite of function above for 4 reasons:

- `Error Handling`: Added a try/catch block to handle any errors during the file search process.
- `Code Simplification`: The mapping of files to items was simplified using the .map function.
- `Sorting`: The sorting now uses a locale-based comparison (localeCompare), which ensures a consistent sort order.
- `Early Exits`: Simplified early return points in the function when no files are found or when no file is selected.

```*.ts
type ManifestQuickPickItem = vscode.QuickPickItem & { filePath: string };

async function getManifestFile(): Promise<string | undefined> {
    try {
        // Find all YAML files in the workspace, excluding node_modules
        const files = await vscode.workspace.findFiles("**/*.yaml", "**/node_modules/**");
        
        // If no files are found, show a warning and exit
        if (files.length === 0) {
            vscode.window.showWarningMessage("No manifest files found in the workspace.");
            return;
        }

        // Map the found files to QuickPick items
        const items: ManifestQuickPickItem[] = files.map(fileUri => ({
            label: path.basename(fileUri.fsPath),
            description: fileUri.fsPath,
            filePath: fileUri.fsPath
        }));

        // Show the QuickPick to the user to select a file
        const fileSelected = await vscode.window.showQuickPick(items.sort((a, b) => a.label.localeCompare(b.label)), {
            title: "Select YAML",
            placeHolder: "Select manifest to deploy"
        });

        // If no file was selected, show a warning and exit
        if (!fileSelected) {
            vscode.window.showWarningMessage("Manifest file not selected");
            return;
        }

        return fileSelected.filePath;
    } catch (error) {
        vscode.window.showErrorMessage(`Error finding manifest files: ${error.message}`);
    }
}

```

Reply:
Thanks for this! Included in recent push